js/libs/form.js

```diff
@@ -114,31 +114,67 @@ Fliplet.Widget.instance('form-builder', function(data) {
 
     if (fields.length && (data.saveProgress && typeof progress === 'object') || entry) {
       fields.forEach(function(field) {
-        if (entry && entry.data && typeof entry.data[field.name] !== 'undefined' && field.populateOnUpdate !== false) {
-          if (field._type === "flDate") {
-            if (Fliplet.Env.get('platform') === 'web') {
-              field.value = moment(entry.data[field.name]).format('DD MMMM YYYY');
-            } else {
-              field.value = moment(entry.data[field.name]).format('YYYY-MM-DD');
-            }
-          } else if (field._type === "flCheckbox" && !Array.isArray(entry.data[field.name])) {
-            field.value = [];
-          } else if (field._type === 'flImage') {
-            var img = entry.data[field.name];
-            field.value = [];
-            if (Array.isArray(img)) {
-              field.value = img;
-            } else if (typeof img === 'string') {
-              field.value.push(img);
-            }
+        if (entry && entry.data && field.populateOnUpdate !== false) {
+          switch (field._type) {
+            case 'flDate':
+              var regex = /([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))/;
+              field.value = regex.exec(entry.data[field.name]) ? entry.data[field.name] : moment().get().format('YYYY-MM-DD');
```

Comment:
shouldn't we parse the `entry.data[field.name]` with moment?

Reply:
@squallstar 
If we will try to parse threw the moment `entry.data[field.name]` when we get a 0 or 1 it will see this value as valid and give as this [result](https://jsfiddle.net/YaOvdiy/Lh35qgyz/6/).
Therefore regexp is safer in this situation.

Reply:
@YaroslavOvdii an ISO date should be considered valid, e.g. `2019-11-22T14:45:58+01:00` should be parsed with moment as it's a valid date. Likewise, `2019` alone is also valid if you try with moment. It gives a warning but the date is parsed correctly to `2019-01-01`.

Given the value exists, I would prefer parsing it. @tonytlwu what do you think?

Reply:
@tonytlwu @YaroslavOvdii bump ^

Reply:
1. I'm happy with supporting an ISO date string, and extracting the date portion of it as the value, e.g. `2019-11-22T14:45:58+01:00` is parsed as `2019-11-22`. Saving the value through the form will then transform it to `2019-11-22`, essentially dropping the time.
1. I don't think `2019` should be parsed into a date. We should just show today's date. I'm happy to discuss additional requirements for how to cleverly interpret other date string patterns.

Reply:
@tonytlwu @squallstar 
I agreed that supporting an ISO date string is needed.
If I may suggest, what if we will extend regexp and look for ISO format string as well with the date string.
Therefore we shall do 2 regexp for the date field.
We will check if the string is in the date string format or the ISO string format.

Reply:
@YaroslavOvdii Yes, I agree with checking for something like `2019-11-22T14:45:58+01:00` as well as `2019-11-22`.

Reply:
@tonytlwu @squallstar Added checking for the ISO date format in the date block.

Reply:
Happy with this part of the PR.